bootstrap/010_helpers.sql

```diff
@@ -149,3 +149,80 @@ $$
 
   return str;
 $$;
+
+create or replace procedure tools.warehouse_size_as_int(v VARCHAR)
+    returns int
+    language sql
+as
+declare
+    INVALID_WAREHOUSE_SIZE EXCEPTION(-20511, 'No such warehouse size, valid sizes are \'X-Small\', \'Small\', \'Medium\',  \'Large\',  \'X-Large\', \'2X-Large\', \'3X-Large\', \'4X-Large\', \'5X-Large\', \'6X-Large\' ');
+begin
+    case (v)
+      when 'X-Small'  then return 1;
+      when 'Small'    then return 2;
+      when 'Medium'   then return 3;
+      when 'Large'    then return 4;
+      when 'X-Large'  then return 5;
+      when '2X-Large' then return 6;
+      when '3X-Large' then return 7;
+      when '4X-Large' then return 8;
+      when '5X-Large' then return 9;
+      when '6X-Large' then return 10;
+      else raise INVALID_WAREHOUSE_SIZE;
+    end;
+end;
+
+create or replace procedure tools.create_warehouses_with_sizes(prefix string, min_size string, max_size string)
+    returns string
+    language sql
+as
+begin
+    let min int := 0;
+    let max int := 0;
+    call warehouse_size_as_int(:min_size) INTO min;
+    call warehouse_size_as_int(:max_size) INTO max;
+
+    let sql string := 'select column1 as size, column2 as credit
```

Comment:
can we call this size_value or size_numeric instead ?

the credit values are different https://docs.snowflake.com/en/user-guide/warehouses-overview#warehouse-size